### PR TITLE
feat: generate stub automatically

### PR DESCRIPTION
**Pull request by kitsuyaazuma**
## WHAT

This pull request introduces a pre-commit hook configuration to automate the generation of stub files (`.pyi`) for the `blazefl` library.

ref: [Automatic stub generation \(stubgen\) \- mypy 1\.16\.0 documentation](https://mypy.readthedocs.io/en/stable/stubgen.html)

## WHY

The primary motivation for this change is to improve the developer experience and static analysis accuracy. Currently, static type checkers like Pyright report errors such as `Stub file not found for "blazefl.core"` due to the absence of these stub files. By automatically generating and including stub files, we can resolve these errors and provide better type information for library consumers and contributors.

### CODE REVIEW

File: blazefl/contrib/fedavg.pyi [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    device: Incomplete
+    num_clients: Incomplete
+    seed: Incomplete
+    def __init__(self, model_selector: ModelSelector, model_name: str, share_dir: Path, state_dir: Path, dataset: PartitionedDataset, device: str, num_clients: int, epochs: int, batch_size: int, lr: float, seed: int, num_parallels: int) -> None: ...
+    @staticmethod
+    def process_client(path: Path, device: str) -> Path: ...
+    @staticmethod
+    def train(model: torch.nn.Module, model_parameters: torch.Tensor, train_loader: DataLoader, device: str, epochs: int, lr: float) -> FedAvgUplinkPackage: ...
+    @staticmethod
+    def evaulate(model: torch.nn.Module, test_loader: DataLoader, device: str) -> tuple[float, float]: ...
```
[the remark this answers is not in the document]
Comment (kitsuyaazuma) [reply blazefl/contrib/fedavg.pyi:85]: This typo will be corrected in a future PR.
